### PR TITLE
Upgrading to NUnit4 including all required dependencies

### PR DESCRIPTION
**Pull request by danielmarbach**
This PR contains all changes related to NUnit4 upgrade including all required dependencies.

The following packages are checked for necessary upgrades but might not necessarily be in the change set if they are not present:
- NServiceBus
- NServiceBus.AcceptanceTesting
- NServiceBus.AcceptanceTests.Sources
- NServiceBus.TransportTests.Sources
- NUnit

In addition, some manual interventions are done when applicable.

## Plan of action

- [x] ⚠️ Build the solution and resolve any remaining errors or warnings
- [x] ✅ Self-review the changes
- [x] 🏗 Set the PR to _In progress_ on the board

### CODE REVIEW

**Comment by danielmarbach** [comment]
Requires https://github.com/Particular/NServiceBus.Persistence.Sql/pull/1547 and then a rebase

**Comment by danielmarbach** [comment]
Rebased

File: src/MsSqlMicrosoftDataClientAcceptanceTests/MsSqlMicrosoftDataClientAcceptanceTests.csproj [hunk 1]
```diff
@@ -14,8 +14,8 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/MsSqlMicrosoftDataClientAcceptanceTests/MsSqlMicrosoftDataClientAcceptanceTests.csproj:18]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/MsSqlMicrosoftDataClientSqlTransportAcceptanceTests/MsSqlMicrosoftDataClientSqlTransportAcceptanceTests.csproj [hunk 2]
```diff
@@ -13,9 +13,9 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" GeneratePathProperty="true" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" GeneratePathProperty="true" />
     <PackageReference Include="NServiceBus.Transport.SqlServer" Version="8.1.3" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/MsSqlMicrosoftDataClientSqlTransportAcceptanceTests/MsSqlMicrosoftDataClientSqlTransportAcceptanceTests.csproj:18]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/MsSqlSystemDataClientAcceptanceTests/MsSqlSystemDataClientAcceptanceTests.csproj [hunk 3]
```diff
@@ -14,8 +14,8 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/MsSqlSystemDataClientAcceptanceTests/MsSqlSystemDataClientAcceptanceTests.csproj:18]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/MySqlAcceptanceTests/MySqlAcceptanceTests.csproj [hunk 4]
```diff
@@ -13,8 +13,8 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" GeneratePathProperty="true" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" GeneratePathProperty="true" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/MySqlAcceptanceTests/MySqlAcceptanceTests.csproj:17]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/OracleAcceptanceTests/OracleAcceptanceTests.csproj [hunk 5]
```diff
@@ -13,8 +13,8 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" GeneratePathProperty="true" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" GeneratePathProperty="true" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/OracleAcceptanceTests/OracleAcceptanceTests.csproj:17]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/PostgreSqlAcceptanceTests/PostgreSqlAcceptanceTests.csproj [hunk 6]
```diff
@@ -13,8 +13,8 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/PostgreSqlAcceptanceTests/PostgreSqlAcceptanceTests.csproj:17]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/PostgreSqlTransportAcceptanceTests/PostgreSqlTransportAcceptanceTests.csproj [hunk 7]
```diff
@@ -13,9 +13,9 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.1.1" GeneratePathProperty="true" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="9.2.2" GeneratePathProperty="true" />
     <PackageReference Include="NServiceBus.Transport.PostgreSql" Version="8.1.3" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/PostgreSqlTransportAcceptanceTests/PostgreSqlTransportAcceptanceTests.csproj:18]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj [hunk 8]
```diff
@@ -16,8 +16,8 @@
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
     <PackageReference Include="Microsoft.SqlServer.DACFx" Version="162.3.566" />
-    <PackageReference Include="NServiceBus" Version="9.1.1" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus" Version="9.2.2" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj:20]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj [hunk 9]
```diff
@@ -14,7 +14,7 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj:17]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/SqlPersistence.PersistenceTests/SqlPersistence.PersistenceTests.csproj [hunk 10]
```diff
@@ -16,8 +16,8 @@
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
-    <PackageReference Include="NServiceBus.PersistenceTests.Sources" Version="9.1.1" GeneratePathProperty="true" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus.PersistenceTests.Sources" Version="9.2.2" GeneratePathProperty="true" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/SqlPersistence.PersistenceTests/SqlPersistence.PersistenceTests.csproj:20]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/SqlPersistence.Tests/SqlPersistence.Tests.csproj [hunk 11]
```diff
@@ -16,8 +16,8 @@
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
-    <PackageReference Include="NServiceBus" Version="9.1.1" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NServiceBus" Version="9.2.2" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/SqlPersistence.Tests/SqlPersistence.Tests.csproj:20]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/TestHelper/TestHelper.csproj [hunk 12]
```diff
@@ -10,7 +10,7 @@
     <PackageReference Include="Microsoft.Data.SqlClient" Version="5.2.1" />
     <PackageReference Include="MySql.Data" Version="9.0.0" />
     <PackageReference Include="Npgsql" Version="8.0.3" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/TestHelper/TestHelper.csproj:13]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/TransactionalSession.MsSqlMicrosoftDataClient.AcceptanceTests/TransactionalSession.MsSqlMicrosoftDataClient.AcceptanceTests.csproj [hunk 13]
```diff
@@ -16,9 +16,9 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTesting" Version="9.1.1" />
+    <PackageReference Include="NServiceBus.AcceptanceTesting" Version="9.2.2" />
     <PackageReference Include="NServiceBus.TransactionalSession" Version="3.1.0" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/TransactionalSession.MsSqlMicrosoftDataClient.AcceptanceTests/TransactionalSession.MsSqlMicrosoftDataClient.AcceptanceTests.csproj:21]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/TransactionalSession.MsSqlSystemDataClient.AcceptanceTests/TransactionalSession.MsSqlSystemDataClient.AcceptanceTests.csproj [hunk 14]
```diff
@@ -16,9 +16,9 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
-    <PackageReference Include="NServiceBus.AcceptanceTesting" Version="9.1.1" />
+    <PackageReference Include="NServiceBus.AcceptanceTesting" Version="9.2.2" />
     <PackageReference Include="NServiceBus.TransactionalSession" Version="3.1.0" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/TransactionalSession.MsSqlSystemDataClient.AcceptanceTests/TransactionalSession.MsSqlSystemDataClient.AcceptanceTests.csproj:21]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

File: src/TransactionalSession.Tests/TransactionalSession.Tests.csproj [hunk 15]
```diff
@@ -14,9 +14,9 @@
     <PackageReference Include="GitHubActionsTestLogger" Version="2.4.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
-    <PackageReference Include="NServiceBus" Version="9.1.1" />
+    <PackageReference Include="NServiceBus" Version="9.2.2" />
     <PackageReference Include="NServiceBus.TransactionalSession" Version="3.1.0" />
-    <PackageReference Include="NUnit" Version="3.14.0" />
+    <PackageReference Include="NUnit" Version="4.2.0" />
```
Comment (cquirosj) [inline src/TransactionalSession.Tests/TransactionalSession.Tests.csproj:19]:
```suggestion
    <PackageReference Include="NUnit" Version="4.2.1" />
```

**Review by cquirosj — COMMENTED** [review]
Updating to NUnit 4.2.1

**Review by lailabougria — APPROVED** [review]

**Pull request merged**